### PR TITLE
fix: apply auth from global settings to repositories and use ~/.m2/se…

### LINKED ISSUE #1504: Regression with GitHub repositories authentication
With JBang 0.97.0, I got the public GitHub repositories authentication working by:
- pushing the GITHUB_TOKEN secret to the action
- injecting it as an env var
- name the GitHub repository where my action is `github`
- the default `~/.m2/settings.xml` for the `github` pushes the secret as the password:

```xml
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <servers>
    <server>
      <id>github</id>
      <username>${env.GITHUB_ACTOR}</username>
      <password>${env.GITHUB_TOKEN}</password>
    </server>
  </servers>
</settings>
```

With the latest version, this setup no longer works and I get a 401 Unauthorized error when JBang tries to get the artifact from the GitHub (not so) public repository.

The setup of the action:
https://github.com/gsmet/github-action-playground/blob/main/action.yml
(see how I installed JBang 0.97.0 or latest via SDKMAN!)

The workflow using the action:
https://github.com/gsmet/github-action-playground-target/blob/main/.github/workflows/publish-image.yml#L39-L44

A working workflow using 0.97.0 (only the Linux build is supposed to work as my dirty SDKMAN! install only works on Linux):
https://github.com/gsmet/github-action-playground-target/actions/runs/3395475566/jobs/5645406665

A failing workflow using latest:
https://github.com/gsmet/github-action-playground-target/actions/runs/3395508036/jobs/5645479380

### PR DESCRIPTION
**Pull request by maxandersen**
…ttings.xml as global rather than user settings

fixes #1504

### CODE REVIEW

**Comment by maxandersen** [comment]
not fully tested yet but this should in *theory* fix it - at least now the auth handler is configured with settings.xml username/passwords.

File: src/main/java/dev/jbang/dependencies/ArtifactResolver.java [hunk 1]
```diff
@@ -155,11 +155,13 @@ private ArtifactResolver(Builder builder) {
...
 		configureSession(session, settings);
+		List<RemoteRepository> partialRepos;
 		if (builder.repositories != null) {
-			repositories = builder.repositories.stream().map(this::toRemoteRepo).collect(Collectors.toList());
+			partialRepos = builder.repositories.stream().map(this::toRemoteRepo).collect(Collectors.toList());
 		} else {
-			repositories = newRepositories();
+			partialRepos = newRepositories();
 		}
+		repositories = system.newResolutionRepositories(session, partialRepos);
```
Comment (quintesse) [inline src/main/java/dev/jbang/dependencies/ArtifactResolver.java:164]: Great find! Where did you encounter this solution?

File: src/main/java/dev/jbang/dependencies/ArtifactResolver.java [hunk 2]
```diff
@@ -354,13 +357,13 @@ private Settings newEffectiveSettings() {
...
-				if (Files.exists(userSettings)) {
-					settingsFile = userSettings.toAbsolutePath();
+				Path globalSettings = Paths.get(System.getProperty("user.home"), ".m2", "settings.xml");
+				if (Files.exists(globalSettings)) {
+					settingsFile = globalSettings.toAbsolutePath();
 				}
 			}
 			if (settingsFile != null) {
-				settingsBuilderRequest.setUserSettingsFile(settingsFile.toFile());
+				settingsBuilderRequest.setGlobalSettingsFile(settingsFile.toFile());
```
Comment (quintesse) [inline src/main/java/dev/jbang/dependencies/ArtifactResolver.java:366]: What exactly is the difference between using global and user?

**Review by quintesse — APPROVED** [review]
LGTM

File: src/main/java/dev/jbang/dependencies/ArtifactResolver.java (hunk 1 above)
Comment (maxandersen) [reply src/main/java/dev/jbang/dependencies/ArtifactResolver.java:164]: Found by doing find references until I found where repos auth gets set. 

So...pure luck? :)

File: src/main/java/dev/jbang/dependencies/ArtifactResolver.java (hunk 2 above)
Comment (maxandersen) [reply src/main/java/dev/jbang/dependencies/ArtifactResolver.java:366]: In our specific currrent situation probably not much but thought might as well just treat home dir settings as global settings if we ever want to introduce using user settings That can override global settings.

File: src/main/java/dev/jbang/dependencies/ArtifactResolver.java (hunk 1 above)
Comment (quintesse) [reply src/main/java/dev/jbang/dependencies/ArtifactResolver.java:164]: Well obviously a bit more than luck. A bit like stumbling around in the dark but at least you brought one of those tiny keychain flashlights 😄

**Pull request merged**